contributors/design-proposals/kubelet-authorizer.md

```diff
@@ -0,0 +1,119 @@
+
+The first step is to identify whether a particular API request is being made by 
+a node, and if so, from which node.
+
+The proposed node authorizer and admission plugin will take a `NodeIdentifier` interface:
+
+```go
+type NodeIdentifier interface {
+  // IdentifyNode determines node information from the given user.Info.
+  // nodeName is the name of the Node API object associated with the user.Info,
+  // and may be empty if a specific node cannot be determined.
+  // isNode is true if the user.Info represents an identity issued to a node.
+  IdentifyNode(user.Info) (nodeName string, isNode bool)
+}
+```
+
+The default `NodeIdentifier` implementation:
+* `isNode` - true if the user groups contain the `system:nodes` group
+* `nodeName` - populated if `isNode` is true, and the user name is in the format `system:node:<nodeName>`
+
+## Node authorizer
+
+A new node authorizer will be inserted into the authorization chain:
+* API server authorizer (authorizes "loopback" API clients used by components within the API server)
+* Node authorizer (new)
+* User-configured authorizers... (e.g. ABAC, RBAC, Webhook)
+
+The node authorizer is responsible for ensuring:
+* The request is made by a node (`IdentifyNode()` returns isNode=true)
+* The request is allowed by the rules contained in the default `system:node` cluster role
+
+Requests that do not meet those conditions are forbidden by this authorizer.
+Subsequent authorizers in the chain can run and choose to allow the request.
+
+Requests from identifiable nodes (`IdentifyNode()` returns nodeName != "") for secrets and configmaps are further restricted:
+* Requests for secrets are limited to `get`, and the requested secret must be related to the requesting node by one of the following relationships:
+  * node -> pod -> secret
+  * node -> pod -> pvc -> pv -> secret
+  * node -> pod -> pvc -> storageclass -> secret
+* Requests for configmaps are limited to `get`, and the requested configmap must be related to the requesting node by one of the following relationships:
+  * node -> pod -> configmap
+
+## Node admission
+
+A new node admission plugin is made available that enforces the following:
+
+Limits `create` of node resources by identifiable nodes:
+* only allow the node object corresponding to the node making the API request
+
+Limits `create` of pod resources by identifiable nodes:
+* only allow pods with mirror pod annotations
+* only allow pods with nodeName set to the node making the API request
+* only allow pods with no secret references
+
+Limits `update`,`patch`,`delete` of node and nodes/status resources by identifiable nodes:
+* only allow modifying the node object corresponding to the node making the API request
+
+Limits `update`,`patch`,`delete` of pod and pod/status resources by identifiable nodes:
+* only allow modifying pods with nodeName set to the node making the API request (requires fetching the pod on delete)
+* do not allow removing a mirror pod annotation
+
```

Comment:
Also create/update/patch of events, limited to `InvolvedObject`s bound to the node